app/src/main/java/ch/epfl/cs311/wanderwave/model/remote/BeaconConnection.kt

```diff
@@ -144,29 +140,16 @@ class BeaconConnection(
     return dataFlow.filterNotNull()
   }
 
-  override fun itemToMap(beacon: Beacon): Map<String, Any> {
-    val beaconMap: HashMap<String, Any> =
-        hashMapOf(
-            "id" to beacon.id,
-            "location" to beacon.location.toMap(),
-            "tracks" to
-                beacon.profileAndTrack.map { profileAndTrack ->
-                  hashMapOf(
-                      "creator" to
-                          db.collection("users")
-                              .document(profileAndTrack.profile?.firebaseUid ?: ""),
-                      "track" to db.collection("tracks").document(profileAndTrack.track.id))
-                })
-    return beaconMap
-  }

```

Comment:
Since the toMap is firebase specific, I would still prefer these methods in the firebase part and not the dataclasses, but probably for another PR.

Reply:
We would need another file for that I think, or maybe put it in some kind of subclass of the firebase connection